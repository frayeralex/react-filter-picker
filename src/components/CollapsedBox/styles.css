.CollapsedBox {
  display: block;
}

.CollapsedBox_expanded .CollapsedBox_content {
  height: auto;
}

.CollapsedBox_header {
  padding: 10px;
  background: rgba(0,0,0,.1);
  display: flex;
  border-top: 1px solid #000;
}

.CollapsedBox_content {
  display: flex;
  flex-wrap: wrap;
  overflow: hidden;
  height: 0;
}

.CollapsedBox_title {
  flex-grow: 1;
}

.CollapsedBox_indicator {
  font-size: 20px;
  font-weight: 600;
}
