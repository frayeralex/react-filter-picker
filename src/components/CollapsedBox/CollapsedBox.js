import React, { PureComponent } from 'react'
import classNames from 'classnames'
import styles from './styles.css'


class CollapsedBox extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      expanded: props.expanded,
    }
  }

  handleHeaderClick = () => {
    this.setState({ expanded: !this.state.expanded });
  }

  render() {
    return (
      <div className={classNames(styles.CollapsedBox, {
        [styles.CollapsedBox_expanded]: this.state.expanded
      })}>
        {this.props.title && (
          <header
            onClick={this.handleHeaderClick}
            className={styles.CollapsedBox_header}>
            <span className={styles.CollapsedBox_title}>{this.props.title}</span>
            <span className={styles.CollapsedBox_indicator}>{this.state.expanded ? '-' : '+'}</span>
          </header>
        )}

        <div className={styles.CollapsedBox_content}>
          {this.props.children}
        </div>
      </div>
    )
  }
}

export default CollapsedBox;
