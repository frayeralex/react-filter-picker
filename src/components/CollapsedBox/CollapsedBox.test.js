import React from 'react';
import CollapsedBox from './CollapsedBox';

describe('<CollapsedBox/>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<CollapsedBox />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should show title correctly', () => {
    const wrapper = shallow(<CollapsedBox />);
    const title = 'title';
    wrapper.setProps({ title });
    expect(
      wrapper
        .find('header span')
        .first()
        .text(),
    ).toEqual(title);
    expect(wrapper).toMatchSnapshot();
  });
  it('should show expanded indicator correctly', () => {
    const wrapper = shallow(<CollapsedBox title="title" />);
    expect(
      wrapper
        .find('header span')
        .at(1)
        .text(),
    ).toEqual('+');
    wrapper.setState({ expanded: true });
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find('header span')
        .at(1)
        .text(),
    ).toEqual('-');
  });
  it('should render children in correct place', () => {
    const wrapper = shallow(<CollapsedBox title="title" />);
    wrapper.setProps({ children: <span className="children">children</span> });
    expect(
      wrapper
        .find('.children')
        .at(0)
        .text(),
    ).toEqual('children');
    expect(wrapper).toMatchSnapshot();
  });
});
