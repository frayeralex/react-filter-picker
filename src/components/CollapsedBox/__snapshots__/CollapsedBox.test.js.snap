// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapsedBox/> should render children in correct place 1`] = `
<div
  className=""
>
  <header
    onClick={[Function]}
  >
    <span>
      title
    </span>
    <span>
      +
    </span>
  </header>
  <div>
    <span
      className="children"
    >
      children
    </span>
  </div>
</div>
`;

exports[`<CollapsedBox/> should render without crashing 1`] = `
<div
  className=""
>
  <div />
</div>
`;

exports[`<CollapsedBox/> should show expanded indicator correctly 1`] = `
<div
  className="undefined"
>
  <header
    onClick={[Function]}
  >
    <span>
      title
    </span>
    <span>
      -
    </span>
  </header>
  <div />
</div>
`;

exports[`<CollapsedBox/> should show title correctly 1`] = `
<div
  className=""
>
  <header
    onClick={[Function]}
  >
    <span>
      title
    </span>
    <span>
      +
    </span>
  </header>
  <div />
</div>
`;
